src/NATS.Client.Core/NatsConnection.cs

```diff
@@ -561,8 +566,40 @@ private NatsUri FixTlsHost(NatsUri uri)
 
     private async Task WaitWithJitterAsync()
     {
+        int retry;
+        TimeSpan backoff;
+        lock (_gate)
+        {
+            retry = _connectRetry++;
+
+            if (Opts.ReconnectWait >= Opts.ReconnectWaitMax)
+            {
+                _backoff = Opts.ReconnectWait;
+            }
+            else if (_backoff == TimeSpan.Zero)
+            {
+                _backoff = Opts.ReconnectWait;
+            }
+            else if (_backoff == Opts.ReconnectWaitMax)
+            {
+            }
+            else
+            {
+                _backoff *= 2;
```

Comment:
Should probably make this something like:

```
                Math.Max(_backoff *= 2, TimeSpan.FromSeconds(1));
```

Because if _backoff is 0 or negative it'll never grow towards ReconnectWaitMax

Reply:
Or maybe just check if _backoff <=0 and if so set it to 1 second... The Math.Max might be too aggressive for smaller values

Reply:
done